File: packages/backend-python/src/schemas/arbitration.py
"""
仲裁案件相关的Pydantic模式
"""

from datetime import datetime
from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field


class ArbitrationStatus(str, Enum):
    """仲裁状态枚举"""

    PENDING = "pending"
    IN_REVIEW = "in_review"
    RESOLVED = "resolved"
    REJECTED = "rejected"


class AnalysisResult(BaseModel):
    """分析结果"""

    analysis: str
    confidence: float = Field(ge=0.0, le=1.0)
    reasoning: str


class SentimentAnalysis(BaseModel):
    """情感分析结果"""

    sentiment: str
    score: float = Field(ge=0.0, le=1.0)
    reasoning: str


class ArbitrationCase(BaseModel):
    """仲裁案件"""

    case_id: str
    report_type: str
    target_code: Optional[str] = None
    qwen_analysis: Optional[AnalysisResult] = None
    doubao_analysis: Optional[SentimentAnalysis] = None
    disagreement_score: float = Field(ge=0.0, le=1.0)
    status: ArbitrationStatus
    created_at: datetime
    updated_at: Optional[datetime] = None
    human_decision: Optional[str] = None
    human_reasoning: Optional[str] = None

    class Config:
        from_attributes = True
        arbitrary_types_allowed = True


class ArbitrationCaseCreate(BaseModel):
    """创建仲裁案件请求"""

    report_type: str
    target_code: Optional[str] = None
    qwen_analysis: Optional[AnalysisResult] = None
    doubao_analysis: Optional[SentimentAnalysis] = None


class ArbitrationCaseUpdate(BaseModel):
    """更新仲裁案件请求"""

    status: Optional[ArbitrationStatus] = None
    human_decision: Optional[str] = None
    human_reasoning: Optional[str] = None


class ArbitrationCaseResponse(BaseModel):
    """仲裁案件响应"""

    success: bool
    message: str
    data: Optional[ArbitrationCase] = None


class ArbitrationCaseListResponse(BaseModel):
    """仲裁案件列表响应"""

    success: bool
    message: str
    data: list[ArbitrationCase]
    total: int
    page: int
    size: int

File: packages/frontend-main/src/services/arbitration.ts
import { request } from './http'

// 仲裁相关类型定义
export interface ArbitrationCase {
  case_id: string
  report_type: string
  target_code: string
  qwen_analysis: {
    analysis: string
    confidence: number
    reasoning: string
  }
  doubao_analysis: {
    sentiment: string
    score: number
    reasoning: string
  }
  disagreement_score: number
  status: string
  consensus_summary?: string
  conflict_summary?: string
  priority_score: number
  created_at: string
  updated_at: string
  human_decision?: string
  human_reasoning?: string
}

export interface ArbitrationCaseListResponse {
  success: boolean
  message: string
  data: ArbitrationCase[]
  total: number
  page: number
  size: number
}

export interface ArbitrationStatistics {
  pendingCases: number
  arbitratedCases: number
  ignoredCases: number
  totalCases: number
}

export interface ArbitrationFeedback {
  decision: 'agree_qwen' | 'agree_doubao' | 'disagree_both' | 'ignore'
  reasoning: string
  confidence: number
  error_type?: string
  additional_notes?: string
}

// 仲裁API服务
export const arbitrationApi = {
  // 获取仲裁案件列表
  getCases: (params: {
    page?: number
    limit?: number
    status?: string
    stockCode?: string
    startDate?: string
    endDate?: string
    sortBy?: string
    sortOrder?: string
  } = {}): Promise<ArbitrationCaseListResponse> => {
    return request.get<ArbitrationCaseListResponse>('/admin/arbitration-cases', {
      params,
      showLoading: true,
    })
  },

  // 获取案件详情
  getCaseDetail: (caseId: string): Promise<ArbitrationCase> => {
    return request.get<ArbitrationCase>(`/admin/arbitration-cases/${caseId}`, {
      showLoading: true,
    })
  },

  // 提交仲裁反馈
  submitFeedback: (caseId: string, feedback: ArbitrationFeedback): Promise<{ success: boolean; message: string }> => {
    return request.post<{ success: boolean; message: string }>(`/admin/arbitration-cases/${caseId}/feedback`, feedback, {
      showLoading: true,
    })
  },

  // 获取统计信息
  getStatistics: (): Promise<ArbitrationStatistics> => {
    return request.get<ArbitrationStatistics>('/admin/arbitration-cases/statistics', {
      showLoading: true,
    })
  },
}
